pkg/pxc/service.go

```diff
@@ -349,6 +349,10 @@ func NewServiceHAProxy(cr *api.PerconaXtraDBCluster) *corev1.Service {
 		)
 	}
 
+	if cr.Spec.HAProxy.Annotations["percona.com/headless-service"] == "true" && svcType == corev1.ServiceTypeClusterIP {
```

Comment:
We need to have consistency. That is why we need to have the same for replicasservice and ProxySQL service.

Reply:
Should I add `Annotations` for them as well?

Reply:
Hi @vishalanarase, yes please add `Annotations` for them.

Reply:
It's `ServiceAnnotations` right?

Reply:
yes